clients/drcachesim/simulator/caching_device.cpp

```diff
@@ -408,7 +410,15 @@ caching_device_t::propagate_eviction(addr_t tag, const caching_device_t *request
         insert_tag(tag, /*is_write=*/false, way, block_idx);
         // Notify the cache policy as if this were a new access to the newly
         // inserted line.
-        access_update(block_idx, way, true /*hit*/);
+        if (prev_serviced_tags_.erase(tag)) {
```

Comment:
The tag should be removed from prev_serviced_tags_ also if it was invalidated in the child cache, right?

I see unit_test_cache_srrip_exclusive doesn't perform any invalidations. Let's add some to ensure this case is tested.